src/Icons/src/Command/ImportIconCommand.php

```diff
@@ -65,33 +63,57 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
             [$fullName, $prefix, $name] = $matches;
 
-            $io->comment(\sprintf('Importing %s...', $fullName));
+            $prefixIcons[$prefix] ??= [];
+            $prefixIcons[$prefix][$name] = $fullName;
+        }
+
+        foreach ($prefixIcons as $prefix => $icons) {
 
-            try {
-                $iconSvg = $this->iconify->fetchIcon($prefix, $name)->toHtml();
-            } catch (IconNotFoundException $e) {
-                $io->error($e->getMessage());
+            if (!$this->iconify->hasIconSet($prefix)) {
+                $io->error(\sprintf('Icon set "%s" not found.', $prefix));
                 $result = Command::FAILURE;
 
                 continue;
             }
 
-            $cursor = new Cursor($output);
-            $cursor->moveUp(2);
-
-            $this->registry->add(\sprintf('%s/%s', $prefix, $name), $iconSvg);
-
-            $license = $this->iconify->metadataFor($prefix)['license'];
+            $metadata = $this->iconify->metadataFor($prefix);
+            $io->writeln([
+                "\n",
+                \sprintf(' IconSet: <fg=bright-white;bg=black>%s</> - %s', $metadata['name'], $prefix),
+                \sprintf('  (Licence: %s)', $metadata['license']['title']),
+                "\n",
+            ]);
+
+            foreach (array_chunk($icons, 25, true) as $iconBatch) {
+                foreach ($iconBatch as $name => $fullName) {
+                    $io->writeln(\sprintf('  Importing %s:%s ...', $prefix, $name));
+                }
+
+                try {
+                    $batchResults = $this->iconify->fetchIcons($prefix, $names);
```

Comment:
In this call ... aren't you fetching all icons at once? The `$names` variable contains ALL icon names, right? Shouldn't you create a new variable with up to 25 icons, which is what you want to download thanks to the previous `array_chunk()`?

Reply:
Good catch @javiereguiluz!  

This should indeed be array_keys($iconBatch) here. 

Will fix when the other one is merged (need to rebase anyway)

Thanks!